From patchwork Wed Mar 18 14:32:53 2026
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Alexey Charkov <alchark@flipper.net>
X-Patchwork-Id: 14481759
Received: from mail-wm1-f53.google.com (mail-wm1-f53.google.com
 [209.85.128.53])
	(using TLSv1.2 with cipher ECDHE-RSA-AES128-GCM-SHA256 (128/128 bits))
	(No client certificate requested)
	by smtp.subspace.kernel.org (Postfix) with ESMTPS id 0DFE53DA7CC
	for <linux-usb@vger.kernel.org>; Wed, 18 Mar 2026 14:33:15 +0000 (UTC)
Authentication-Results: smtp.subspace.kernel.org;
 arc=none smtp.client-ip=209.85.128.53
ARC-Seal: i=1; a=rsa-sha256; d=subspace.kernel.org; s=arc-20240116;
	t=1773844398; cv=none;
 b=OW8j134NuhbTRxgZlJ6io3JAurtvinWmI7GUTc6kHZIMWl1Wq5BSTYKQEcOtPHf+fAnJPdvaTOravY6MQ3jynp2sdlZcPnrTQZM0Ax7daOnI//MfEOB5f96apVrm2nkesinvUOMTwTyLgZOT9v42vXqMrUj1wpleVYbhPWLdE6g=
ARC-Message-Signature: i=1; a=rsa-sha256; d=subspace.kernel.org;
	s=arc-20240116; t=1773844398; c=relaxed/simple;
	bh=T+J1OY8trN/7RzkbAWgyzywsuUmbTSqBleeNAx93u6E=;
	h=From:Date:Subject:MIME-Version:Content-Type:Message-Id:References:
	 In-Reply-To:To:Cc;
 b=o2BrL4xRmHEeLcWaU8h6T3JRxcnepHn4rGXlvEAbUvC/lYV8cxSFYlVPrSWftcLXkBUCd1CmB2cRRDeFdP/feAre7x/pkfgo3ZMd9qnWU1YqJQyyhsJAeo/kGHajLmDy0DJro26qBVbzerfNAwloAIs7XwmtCqcxX/qXYcPk9Jw=
ARC-Authentication-Results: i=1; smtp.subspace.kernel.org;
 dmarc=pass (p=quarantine dis=none) header.from=flipper.net;
 spf=pass smtp.mailfrom=flipper.net;
 dkim=pass (2048-bit key) header.d=flipper.net header.i=@flipper.net
 header.b=hspEZ5/0; arc=none smtp.client-ip=209.85.128.53
Authentication-Results: smtp.subspace.kernel.org;
 dmarc=pass (p=quarantine dis=none) header.from=flipper.net
Authentication-Results: smtp.subspace.kernel.org;
 spf=pass smtp.mailfrom=flipper.net
Authentication-Results: smtp.subspace.kernel.org;
	dkim=pass (2048-bit key) header.d=flipper.net header.i=@flipper.net
 header.b="hspEZ5/0"
Received: by mail-wm1-f53.google.com with SMTP id
 5b1f17b1804b1-48534237460so74031885e9.3
        for <linux-usb@vger.kernel.org>; Wed, 18 Mar 2026 07:33:15 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=flipper.net; s=google; t=1773844394; x=1774449194;
 darn=vger.kernel.org;
        h=cc:to:in-reply-to:references:message-id:content-transfer-encoding
         :mime-version:subject:date:from:from:to:cc:subject:date:message-id
         :reply-to;
        bh=S9sCFNxATE1UQcKX1OKooIck8dS1xOFB94Y0TqD1Dbc=;
        b=hspEZ5/0KXR48ulicGhIDKAuX54iDCKlzIVaYdc+CGbUiF7ZyjQcecZpPsvFiKDGlz
         ueXShCx5A2PsNVX3y+LGkIgFvDNRYjzlYnC7mTpqh9goI/CXX2wbTjE/QilMo4Ow66Sf
         gP+EaO2kuhdhApe6rt7kwAb6NfOU9EvcGLTi+pKG5gMLb9eVhVluRYtUOb0rqyc4OoDN
         keVsjB2yjc0iEeB8RjNCOwnE5fB3Uv+ueKjUjoqDu93SwtXjL7biwUDghehyCxwNkaQt
         iZlbUXOAQwXN+rO54KH78paj54uywEBDsUf2tq7Am9TBZ8lJ1cBd7a3O+xKuOtanEy5A
         mGaw==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20251104; t=1773844394; x=1774449194;
        h=cc:to:in-reply-to:references:message-id:content-transfer-encoding
         :mime-version:subject:date:from:x-gm-gg:x-gm-message-state:from:to
         :cc:subject:date:message-id:reply-to;
        bh=S9sCFNxATE1UQcKX1OKooIck8dS1xOFB94Y0TqD1Dbc=;
        b=GL9Zif2azq8jYgYpcVGvKcvJ8L0KJKceCvY9iu9jBrF70eq639bEv/eZiDM7Ah43s1
         YWzvGmaoCH2TIIDTdZJsyIg+rF2cMM+BjrtYsTGtdzEKboKpZD6DboeN4jXQFD35Cl6q
         oY6seD5MCI7od8ctZ/+nkMT2qOP1auQCbj1G9k2M39aTQFSz+qOm94t04ilu/FVzTbXl
         ygrPs0mmRMrtnsb0pfd5qw1RrWlNLNAPYq1khX2Knq0BMgTF/iqR4/pirYcTIIAh9rOc
         K1BrcxrDMZEfp1vPDTcheca1NAnuJKpuS29X2/4l5PjdC++aoAIqor2gT4XmLO/W4jsZ
         FQTw==
X-Forwarded-Encrypted: i=1;
 AJvYcCX8t9Pgwv2bOuiIrGl1YhK5Z8yHjgyTvOvCvdvx2vIcKFN1WiTt2bXeCCrh1qksx3sYQm9Ep9r/veE=@vger.kernel.org
X-Gm-Message-State: AOJu0YxBoOIB+Zcy0JMs0M5i8xYOvhXpu6lBlehOz9o4NX3gpuhgGKw5
	h0W6HcIuvl2Iy+cTYXfUqtrOZ0UEIUaol1G3WHtIN13Iy9k9zLcQuLGfwUOS0w5Q2s8=
X-Gm-Gg: ATEYQzxPq3AMZAHddApNMqphhD8Vzy3YoYt64lcF2+1Ze2D7yNyjnC3ZU9ftahBjhh6
	bHeNWVGZwoOFrl4z1Oy5bDQGqRLbjBTvMBHX3AByVBoT62iZJv9e5RI5Rh9NVJLK7qdigEFNeyp
	IfMqbGpJd6d3Yf1OqBrZWiXvueR4fR7vlh+zOSjG/1PYLpzMdbQ7mt4e6m5FgeQc8hTqKQj85Pj
	/aVojImnw9P7MPt9gnuUa8vAQsaw8iZ59g5tpy8Ce/K+Rujiqh5rEU30+vKnHRP1UwbMd4Zj7BX
	YfJe+P2cMmA2QhVzEJnxe36qYO2qBpaj0OMC9sQt6XvHFSkQZVT8CPoBrp6Acyr+o2pgoruMMZk
	Y3FFLjA5VyBcvNdL9TA7WVfOkRzdOiB+obJPyxgYh7LeYE1w4k2rqqq2lkKD/cHnr2Lsn60AC9u
	GMBlcLfYNtJ+EiyBAArahVmGP+B6eOVdFvQHXndx9p96bSvivgNB+027DY8HyV/cBaOBEg/VlZZ
	rWUGg==
X-Received: by 2002:a05:600c:3acf:b0:485:4438:6ef6 with SMTP id
 5b1f17b1804b1-486f44435eemr57308705e9.18.1773844394264;
        Wed, 18 Mar 2026 07:33:14 -0700 (PDT)
Received: from alchark-surface.localdomain (bba-86-98-192-109.alshamil.net.ae.
 [86.98.192.109])
        by smtp.gmail.com with ESMTPSA id
 5b1f17b1804b1-4856ea8fb0dsm270642085e9.3.2026.03.18.07.33.11
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Wed, 18 Mar 2026 07:33:13 -0700 (PDT)
From: Alexey Charkov <alchark@flipper.net>
Date: Wed, 18 Mar 2026 18:32:53 +0400
Subject: [PATCH v4 1/4] dt-bindings: vendor-prefixes: Add Hynetek
 Semiconductor Co., Ltd.
Precedence: bulk
X-Mailing-List: linux-usb@vger.kernel.org
List-Id: <linux-usb.vger.kernel.org>
List-Subscribe: <mailto:linux-usb+subscribe@vger.kernel.org>
List-Unsubscribe: <mailto:linux-usb+unsubscribe@vger.kernel.org>
MIME-Version: 1.0
Message-Id: <20260318-husb311-v4-1-69e029255430@flipper.net>
References: <20260318-husb311-v4-0-69e029255430@flipper.net>
In-Reply-To: <20260318-husb311-v4-0-69e029255430@flipper.net>
To: Rob Herring <robh@kernel.org>, Krzysztof Kozlowski <krzk+dt@kernel.org>,
 Conor Dooley <conor+dt@kernel.org>,
 Heikki Krogerus <heikki.krogerus@linux.intel.com>,
 Greg Kroah-Hartman <gregkh@linuxfoundation.org>,
 Gene Chen <gene_chen@richtek.com>, Yuanshen Cao <alex.caoys@gmail.com>
Cc: Sebastian Reichel <sebastian.reichel@collabora.com>,
 devicetree@vger.kernel.org, linux-kernel@vger.kernel.org,
 linux-usb@vger.kernel.org,
 Krzysztof Kozlowski <krzysztof.kozlowski@oss.qualcomm.com>,
 Alexey Charkov <alchark@flipper.net>,
 Conor Dooley <conor.dooley@microchip.com>
X-Mailer: b4 0.14.3
X-Developer-Signature: v=1; a=openpgp-sha256; l=1010; i=alchark@flipper.net;
 h=from:subject:message-id; bh=T+J1OY8trN/7RzkbAWgyzywsuUmbTSqBleeNAx93u6E=;
 b=owGbwMvMwCW2adGNfoHIK0sZT6slMWTu2r7kzYviGyXduTk7ey0XzS+bNjFVv7SWWf9GoGYDk
 /3q59dOdkxkYRDjYrAUU2SZ+22J7VQjvlm7PDy+wsxhZQIZIi3SwAAELAx8uYl5pUY6Rnqm2oZ6
 hoY6xjpGDFycAjDVljsY/keJNMj5lHBFzBWZy3zEW3gKmxuXhvZb35Rme5s7+VdC9jP8j95a/Xv
 zNd/oetNbPyeJZx/YedqnmmtZWuL5OMNdWc942AE=
X-Developer-Key: i=alchark@flipper.net; a=openpgp;
 fpr=9DF6A43D95320E9ABA4848F5B2A2D88F1059D4A5

Hynetek Semiconductor Co., Ltd. focuses on intelligent energy control
technology, mainly for the intelligent fast charging and digital energy
fields.

Link: https://en.hynetek.com/
Acked-by: Conor Dooley <conor.dooley@microchip.com>
Signed-off-by: Alexey Charkov <alchark@flipper.net>
---
 Documentation/devicetree/bindings/vendor-prefixes.yaml | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Documentation/devicetree/bindings/vendor-prefixes.yaml b/Documentation/devicetree/bindings/vendor-prefixes.yaml
index 5a806faf7170..3ffb1a8e24af 100644
--- a/Documentation/devicetree/bindings/vendor-prefixes.yaml
+++ b/Documentation/devicetree/bindings/vendor-prefixes.yaml
@@ -747,6 +747,8 @@ patternProperties:
     description: Hycon Technology Corp.
   "^hydis,.*":
     description: Hydis Technologies
+  "^hynetek,.*":
+    description: Hynetek Semiconductor Co., Ltd.
   "^hynitron,.*":
     description: Shanghai Hynitron Microelectronics Co. Ltd.
   "^hynix,.*":

From patchwork Wed Mar 18 14:32:54 2026
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Alexey Charkov <alchark@flipper.net>
X-Patchwork-Id: 14481760
Received: from mail-wm1-f52.google.com (mail-wm1-f52.google.com
 [209.85.128.52])
	(using TLSv1.2 with cipher ECDHE-RSA-AES128-GCM-SHA256 (128/128 bits))
	(No client certificate requested)
	by smtp.subspace.kernel.org (Postfix) with ESMTPS id F21843DC4CF
	for <linux-usb@vger.kernel.org>; Wed, 18 Mar 2026 14:33:18 +0000 (UTC)
Authentication-Results: smtp.subspace.kernel.org;
 arc=none smtp.client-ip=209.85.128.52
ARC-Seal: i=1; a=rsa-sha256; d=subspace.kernel.org; s=arc-20240116;
	t=1773844401; cv=none;
 b=IiP+0oy1ea3aD6ccSGR08jmx1ZApb7hRaO8WlKXNLXiv5vt9ZCCalC1/RFOex+6IuwwIhmmuZnWY+CUYmCVIa7wi7K4G6ZfHhKaXP67LwPw6JrYlNMJyoyCDJZ4BRBiX45dK6Q8ILqJGSFodyBtCDLhkMnUqBoXDsKGClvdn940=
ARC-Message-Signature: i=1; a=rsa-sha256; d=subspace.kernel.org;
	s=arc-20240116; t=1773844401; c=relaxed/simple;
	bh=Nf4NOywSXVhOZeAxW30T2QFpp+SRZKKoI+FdQwqCvkA=;
	h=From:Date:Subject:MIME-Version:Content-Type:Message-Id:References:
	 In-Reply-To:To:Cc;
 b=Q8hZHZ5kSC/Y9oYXVdP4ZlpZHAqjbA/9YTWWufap0JSUqsz8tMFugJ/sOMeShyyyR36kbAiVKJFHi+YDs9rT97DRLV3C3x1RVKidbrZP5SrCcKOsvKDi4sBalHjhCVW4hyAg6l8ftOyhLyu1Nfnl/B9SFYeDmpuEufkUBSYX43c=
ARC-Authentication-Results: i=1; smtp.subspace.kernel.org;
 dmarc=pass (p=quarantine dis=none) header.from=flipper.net;
 spf=pass smtp.mailfrom=flipper.net;
 dkim=pass (2048-bit key) header.d=flipper.net header.i=@flipper.net
 header.b=VOqz30sv; arc=none smtp.client-ip=209.85.128.52
Authentication-Results: smtp.subspace.kernel.org;
 dmarc=pass (p=quarantine dis=none) header.from=flipper.net
Authentication-Results: smtp.subspace.kernel.org;
 spf=pass smtp.mailfrom=flipper.net
Authentication-Results: smtp.subspace.kernel.org;
	dkim=pass (2048-bit key) header.d=flipper.net header.i=@flipper.net
 header.b="VOqz30sv"
Received: by mail-wm1-f52.google.com with SMTP id
 5b1f17b1804b1-485445e80bdso60073085e9.0
        for <linux-usb@vger.kernel.org>; Wed, 18 Mar 2026 07:33:18 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=flipper.net; s=google; t=1773844397; x=1774449197;
 darn=vger.kernel.org;
        h=cc:to:in-reply-to:references:message-id:content-transfer-encoding
         :mime-version:subject:date:from:from:to:cc:subject:date:message-id
         :reply-to;
        bh=T6FNiGJVrXwKHg5+eSEHDGqgSud+po5BW87ZoAmXM6o=;
        b=VOqz30svDBZg7oLp4UdkfoikCAOrTuyiYXJmmVNPxn73DeOLmObxWFykX0YJRX7AqA
         R+QRuHNzDyjSHGQzjN+IqIusDYbMgerBWPgIH3isJW0zSS4qscRez/jL79xBOaQ9vqU/
         4L/zNJ6B78UECokwbmabPWRvWjwONbnLNYfdo1wBr5uMzXmtS/Ka1tTgCSsq8iB2pJOC
         GhwT9mkmFIIzykZWcdCwCp1AaegpN3KqC+zx4gjB6FDEISVkOYeKHhY0mfai3TI85EFw
         SS6JT4xEm/tOcJo+tAujVjZPUnt1VKKMekiD2i1+g6Ytc/4CGvZP44BniHkZClkANKQw
         GlOg==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20251104; t=1773844397; x=1774449197;
        h=cc:to:in-reply-to:references:message-id:content-transfer-encoding
         :mime-version:subject:date:from:x-gm-gg:x-gm-message-state:from:to
         :cc:subject:date:message-id:reply-to;
        bh=T6FNiGJVrXwKHg5+eSEHDGqgSud+po5BW87ZoAmXM6o=;
        b=PR2TIRvppsGoVBjhS3HAS88Q8QIp1QfSBZilpFXFFfi4s0T2JrMKdvprPTshO75IaZ
         Yh/oOPZpZS2QW8MNB8hO7hcT9nnBnMwDR2MPMGJMk5puQE2Xq5z9XJ6GKNAeJHwlzwbV
         if1e5oG/Ykp5LCvf3fTRSAAFQcDAXY0+X9T1anSM5ckxBTt6J3PdkaunGOwg/TQXKZwy
         viKxplK6GE+eoGVt5tW56Tm5sxuAysPNj0XJASbXnNJ5/yqse7P9yhq8z0ngJqJoprZB
         fMx0TO9KsD2bt3vyzUYRwng/0lRazRzICrQdNaZJpjA9ns+22PSLpxP151JZNG72FYi5
         QI7w==
X-Forwarded-Encrypted: i=1;
 AJvYcCXa7D07YThMw+r6jhNBrN0N104V96NQR+nlV+cD/ao+oCN6aN28rL8Bh4f0AHC4kitYNNqS19sNDss=@vger.kernel.org
X-Gm-Message-State: AOJu0YwUOdL7D2WuPcy5nq2aaOjsagIE1ugl/ocqfmOben7wM5+8oGp2
	EszWXxQ0OnsLsBc/JT3xB75JGHQBrFIe4GgS1eddhXpCXI8ezzcxSJjDjpRtb9/RSvg=
X-Gm-Gg: ATEYQzykbbqT1CRiSf665u1kGrHVRn/Htsmnt6c/PtWnEGAuVn3jviqG3TPlzUJ+Ghl
	o9g71qlRS50rw77YcifTVNAmsPTlO7ZWWEAyyjoM9lBBLlH38a1xBauewU67RS/Bbs8Mzmlz0Aq
	f7e86QPdgoiwd52wGIJweLiv2Pb3wC1tqdJaMgtUOPRrOAf6ulPfCwzxELIJscaa07wAhqYCD6v
	BIrmwjfNYzRM+7T1nd8vAswnMn7SmA/aUUaqKqnkcoQou4RMv7Dv6Pj8BjnVYhmwGm8jSKSW64p
	MkfN+iZQS/MP6kPc3Rlpzy2tbmhlZYe8h5aeEG4o/tRAdhpvQySzy2ohuT56br23wUBJQr2RzRm
	hjrIxuNfLJpHLHqiZoYRBMMzNxNg8Ambu21ruqMtpEvC6SxG7P22NeT8HZJMgMlnIrGr8DYEgCR
	9YJjvM9xB7Ux87J7JCOkn2DdSxcwJs2xJ90Pqpfppcn8bnXALZKLWH52Kdb3xP0B7FlF1Zior3p
	R89X334zJ2nof2m
X-Received: by 2002:a05:600c:4685:b0:485:3c66:e230 with SMTP id
 5b1f17b1804b1-486f458127fmr61968915e9.29.1773844397154;
        Wed, 18 Mar 2026 07:33:17 -0700 (PDT)
Received: from alchark-surface.localdomain (bba-86-98-192-109.alshamil.net.ae.
 [86.98.192.109])
        by smtp.gmail.com with ESMTPSA id
 5b1f17b1804b1-4856ea8fb0dsm270642085e9.3.2026.03.18.07.33.14
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Wed, 18 Mar 2026 07:33:16 -0700 (PDT)
From: Alexey Charkov <alchark@flipper.net>
Date: Wed, 18 Mar 2026 18:32:54 +0400
Subject: [PATCH v4 2/4] dt-bindings: usb: richtek,rt1711h: Switch ETEK
 ET7304 to use a fallback compatible
Precedence: bulk
X-Mailing-List: linux-usb@vger.kernel.org
List-Id: <linux-usb.vger.kernel.org>
List-Subscribe: <mailto:linux-usb+subscribe@vger.kernel.org>
List-Unsubscribe: <mailto:linux-usb+unsubscribe@vger.kernel.org>
MIME-Version: 1.0
Message-Id: <20260318-husb311-v4-2-69e029255430@flipper.net>
References: <20260318-husb311-v4-0-69e029255430@flipper.net>
In-Reply-To: <20260318-husb311-v4-0-69e029255430@flipper.net>
To: Rob Herring <robh@kernel.org>, Krzysztof Kozlowski <krzk+dt@kernel.org>,
 Conor Dooley <conor+dt@kernel.org>,
 Heikki Krogerus <heikki.krogerus@linux.intel.com>,
 Greg Kroah-Hartman <gregkh@linuxfoundation.org>,
 Gene Chen <gene_chen@richtek.com>, Yuanshen Cao <alex.caoys@gmail.com>
Cc: Sebastian Reichel <sebastian.reichel@collabora.com>,
 devicetree@vger.kernel.org, linux-kernel@vger.kernel.org,
 linux-usb@vger.kernel.org,
 Krzysztof Kozlowski <krzysztof.kozlowski@oss.qualcomm.com>,
 Alexey Charkov <alchark@flipper.net>
X-Mailer: b4 0.14.3
X-Developer-Signature: v=1; a=openpgp-sha256; l=1387; i=alchark@flipper.net;
 h=from:subject:message-id; bh=Nf4NOywSXVhOZeAxW30T2QFpp+SRZKKoI+FdQwqCvkA=;
 b=owGbwMvMwCW2adGNfoHIK0sZT6slMWTu2r5k6QO9eaLbS58pb1OVOd8p9c2hpqGCUeXio3mFh
 SdtjYW3dExkYRDjYrAUU2SZ+22J7VQjvlm7PDy+wsxhZQIZIi3SwAAELAx8uYl5pUY6Rnqm2oZ6
 hoY6xjpGDFycAjDVj7cw/A+71R4454XJIUG/v0cVp3zz3F3p9MakofJ9+YH8xAPsu14w/DNpsHX
 UnhE+3+pw5bMcp/fyv+YdOSxx9sqkOb+ijjL7BfIDAA==
X-Developer-Key: i=alchark@flipper.net; a=openpgp;
 fpr=9DF6A43D95320E9ABA4848F5B2A2D88F1059D4A5

As stated in [1], ETEK ET7304 is identical to Richtek RT1715, except for
the VID value in its registers, so reflect it in the bindings via a
fallback compatible.

As there are various TCPCI chips by different vendors reimplementing the
registers and behavior of the RT1711H/RT1715, fallback compatibles will
scale better.

Link: https://lore.kernel.org/all/20260220-et7304-v3-2-ede2d9634957@gmail.com/ [1]
Signed-off-by: Alexey Charkov <alchark@flipper.net>
Reviewed-by: Krzysztof Kozlowski <krzysztof.kozlowski@oss.qualcomm.com>
---
 Documentation/devicetree/bindings/usb/richtek,rt1711h.yaml | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/Documentation/devicetree/bindings/usb/richtek,rt1711h.yaml b/Documentation/devicetree/bindings/usb/richtek,rt1711h.yaml
index 1eb611f35998..210090308e7c 100644
--- a/Documentation/devicetree/bindings/usb/richtek,rt1711h.yaml
+++ b/Documentation/devicetree/bindings/usb/richtek,rt1711h.yaml
@@ -18,10 +18,14 @@ description: |
 
 properties:
   compatible:
-    enum:
-      - etekmicro,et7304
-      - richtek,rt1711h
-      - richtek,rt1715
+    oneOf:
+      - enum:
+          - richtek,rt1711h
+          - richtek,rt1715
+      - items:
+          - enum:
+              - etekmicro,et7304
+          - const: richtek,rt1715
     description:
       RT1711H support PD20, ET7304 and RT1715 support PD30 except Fast Role Swap.
 

From patchwork Wed Mar 18 14:32:55 2026
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Alexey Charkov <alchark@flipper.net>
X-Patchwork-Id: 14481761
Received: from mail-wm1-f46.google.com (mail-wm1-f46.google.com
 [209.85.128.46])
	(using TLSv1.2 with cipher ECDHE-RSA-AES128-GCM-SHA256 (128/128 bits))
	(No client certificate requested)
	by smtp.subspace.kernel.org (Postfix) with ESMTPS id B743F3DD511
	for <linux-usb@vger.kernel.org>; Wed, 18 Mar 2026 14:33:21 +0000 (UTC)
Authentication-Results: smtp.subspace.kernel.org;
 arc=none smtp.client-ip=209.85.128.46
ARC-Seal: i=1; a=rsa-sha256; d=subspace.kernel.org; s=arc-20240116;
	t=1773844403; cv=none;
 b=DvI4MMche5QLmpuhLCXMGpRqE4Sm5zAa/X2uUk+Ed/xj5D8FIUEt8+N7F/TY1On47tpNaOPYqdOMaW/C4Oererkj+rK5H/wixur0wmrKLckXUUEUZfToSU0If0Dqsq0HLdJiBr6IcCKhf+u+xwDqjlW4l6Z6tV3Aq1E/MZ8Ksa4=
ARC-Message-Signature: i=1; a=rsa-sha256; d=subspace.kernel.org;
	s=arc-20240116; t=1773844403; c=relaxed/simple;
	bh=SWwU2pOPUl8qRUMi6Zu0uI8G4Sbh5J1V/0WUKZlPYKM=;
	h=From:Date:Subject:MIME-Version:Content-Type:Message-Id:References:
	 In-Reply-To:To:Cc;
 b=OhVmfmLY+XKMybQYg4u5FftwdGCfNtfSo2V7uo1UM6Wj5k4P5E86p7sU3GFMeYZ/1TGQYQhUAQnYVM5k6cAiZs0O+RVjyYqlHVssrr7N9knYP1q3pwLN99yxKc4N7mcZoVxNBVeF68EO53rbZ22IEp63joQY++Qc2fA5Xn7/Ry4=
ARC-Authentication-Results: i=1; smtp.subspace.kernel.org;
 dmarc=pass (p=quarantine dis=none) header.from=flipper.net;
 spf=pass smtp.mailfrom=flipper.net;
 dkim=pass (2048-bit key) header.d=flipper.net header.i=@flipper.net
 header.b=eXUceUfZ; arc=none smtp.client-ip=209.85.128.46
Authentication-Results: smtp.subspace.kernel.org;
 dmarc=pass (p=quarantine dis=none) header.from=flipper.net
Authentication-Results: smtp.subspace.kernel.org;
 spf=pass smtp.mailfrom=flipper.net
Authentication-Results: smtp.subspace.kernel.org;
	dkim=pass (2048-bit key) header.d=flipper.net header.i=@flipper.net
 header.b="eXUceUfZ"
Received: by mail-wm1-f46.google.com with SMTP id
 5b1f17b1804b1-48541edecf9so75671475e9.1
        for <linux-usb@vger.kernel.org>; Wed, 18 Mar 2026 07:33:21 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=flipper.net; s=google; t=1773844400; x=1774449200;
 darn=vger.kernel.org;
        h=cc:to:in-reply-to:references:message-id:content-transfer-encoding
         :mime-version:subject:date:from:from:to:cc:subject:date:message-id
         :reply-to;
        bh=eAbEi8TFYqnDaY8fBbFVjb7bhf3BVzUht5+vvpwhfSY=;
        b=eXUceUfZGGd/AwubPq7g4u1OUSJusxFnHx+O451/KHkDynugiQWr59gH+LjjsuMbIT
         ECRakzS2Kq3LYboKBii5Tfj9VthqpgMr0UP3vc8ZeA1BOo2yYKnuWoRY1tHXglq7v3PG
         bXSt4h29NXcxe+AX5pBMXwnMFQN2u08AIw9XtutDAQJCad2Xgw+rAGpB2EV4o750XEL0
         maxsfiHz24e82EBEiWyRu+L/+7ssY4buOgg81jXlJwr8BavDqUFk8eHdpVqdazKt/rNI
         gEyKE4IlQHdTkvW+vvUPqDqLBlfJ+CU/XyMzdgO26atZccp0rgybmtes9ueeAV2nm/cJ
         ek3g==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20251104; t=1773844400; x=1774449200;
        h=cc:to:in-reply-to:references:message-id:content-transfer-encoding
         :mime-version:subject:date:from:x-gm-gg:x-gm-message-state:from:to
         :cc:subject:date:message-id:reply-to;
        bh=eAbEi8TFYqnDaY8fBbFVjb7bhf3BVzUht5+vvpwhfSY=;
        b=YSoQVIQVujlEFgtWuEc75ro4g6wSypq3Mhqra2VDkSMA7EedzwdNG9ET5Qi0FpEMcE
         DqptpIEPARxhgCBf0JFmuIfXRHDu4GNJfCAXbeJ7YoYtcwc8ECi2T0CPjOv+J7OkidKq
         312SiO67b/maV03pvF/wq7cTtfCrt/SHD+UKxCpu4agpbbHqhxtuYpfng37eBOFCLW5o
         JsI6YoWZd8j+87WBYPtqavV3TdxN5frF79RWGEQMfK2pz6s6oxQMkc6Io005g1ikFgC3
         wbKkenXXorZHBPgM8+xA1nATszcxkZMQ4753N8wxivmJvrraRuVK8gdlx0bwZQPNhBrp
         WOwA==
X-Forwarded-Encrypted: i=1;
 AJvYcCWwB08NoaaNSSk/x8vZVOmITi5i8YuyESkN70fykhizCSjjjEEePhKax+gcYAr8nvuvbwIojTo35XU=@vger.kernel.org
X-Gm-Message-State: AOJu0YzFJCx918L0HZ/+rkpvQlFjJ6DIVlfbOJaI/QZqzt4YOcNYcbWg
	v1+Ii/Z5wop3azgbMKqpybVya0Ykd/2ye9YmF5wolyJsjUYFk9XaBlqZ+vM98EUKevQ=
X-Gm-Gg: ATEYQzwehRuTIt8zqLYShyWllK+xRqIBqpNRcDye1Ep0IrjIwh7F/eZnL4LXk578ikH
	SO3R33AX6ephqvtlWTIjMANWm6T0V6JcRFo+fXgI1L/UX8yM7odICx/O1Du5vVK5LgahuZ7Yp7f
	PnAI+0kAUmP5WiLnqvWiIihJl2FqTfzmwvLRlTGhaBwZUcw1W+LJ0C3kz5nluR/oAT+XJeHA3+X
	TAaWg8bOXDrZ7GXA5KubCtMVYykjZz6YGNenJdieeyjx/la1SpaLiZSbJTThIUJNqy8GrRVf6Dk
	sT9YmY4MKzRgD8kcStyKJ7kt9div2Sg3+5QXyOo/X4NkLNN15EjYekVMjV/mEN46xONIqBwIGQp
	r3kCtik2pHorCdA60WHz5EQMt87ASb3VIna4v9Hbba4CMv+K7HmaemZVdS9k0SR5kMi8OqoSZVS
	AT76BihMablkFeiVV2han+MRcLL2xfgWyLSk/sVcWUu84pQuqj6FUHzQ4sa5jyZ9dGTkl6PugAr
	Ch5gQ==
X-Received: by 2002:a05:600c:34d5:b0:485:40fd:8390 with SMTP id
 5b1f17b1804b1-486f4457b0fmr57784475e9.26.1773844399943;
        Wed, 18 Mar 2026 07:33:19 -0700 (PDT)
Received: from alchark-surface.localdomain (bba-86-98-192-109.alshamil.net.ae.
 [86.98.192.109])
        by smtp.gmail.com with ESMTPSA id
 5b1f17b1804b1-4856ea8fb0dsm270642085e9.3.2026.03.18.07.33.17
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Wed, 18 Mar 2026 07:33:19 -0700 (PDT)
From: Alexey Charkov <alchark@flipper.net>
Date: Wed, 18 Mar 2026 18:32:55 +0400
Subject: [PATCH v4 3/4] dt-bindings: usb: richtek,rt1711h: Add Hynetek
 HUSB311
Precedence: bulk
X-Mailing-List: linux-usb@vger.kernel.org
List-Id: <linux-usb.vger.kernel.org>
List-Subscribe: <mailto:linux-usb+subscribe@vger.kernel.org>
List-Unsubscribe: <mailto:linux-usb+unsubscribe@vger.kernel.org>
MIME-Version: 1.0
Message-Id: <20260318-husb311-v4-3-69e029255430@flipper.net>
References: <20260318-husb311-v4-0-69e029255430@flipper.net>
In-Reply-To: <20260318-husb311-v4-0-69e029255430@flipper.net>
To: Rob Herring <robh@kernel.org>, Krzysztof Kozlowski <krzk+dt@kernel.org>,
 Conor Dooley <conor+dt@kernel.org>,
 Heikki Krogerus <heikki.krogerus@linux.intel.com>,
 Greg Kroah-Hartman <gregkh@linuxfoundation.org>,
 Gene Chen <gene_chen@richtek.com>, Yuanshen Cao <alex.caoys@gmail.com>
Cc: Sebastian Reichel <sebastian.reichel@collabora.com>,
 devicetree@vger.kernel.org, linux-kernel@vger.kernel.org,
 linux-usb@vger.kernel.org,
 Krzysztof Kozlowski <krzysztof.kozlowski@oss.qualcomm.com>,
 Alexey Charkov <alchark@flipper.net>
X-Mailer: b4 0.14.3
X-Developer-Signature: v=1; a=openpgp-sha256; l=1491; i=alchark@flipper.net;
 h=from:subject:message-id; bh=SWwU2pOPUl8qRUMi6Zu0uI8G4Sbh5J1V/0WUKZlPYKM=;
 b=owGbwMvMwCW2adGNfoHIK0sZT6slMWTu2r6kJebd/O8X46MuTIhZ12Ls2OkgcKAzegJbpcrxK
 v3nvHu9OyawMIhxMViKKbLM/bbEdqoR36xdHh5fYeawMoEMkRZpYAACFga+3MS8UiMdIz1TbUM9
 Q0MdYx0jBi5OAZjq2i0MX6UDZJJX6aeZlGn1vzH2trgR4Ph5x5dZf9csZ5bOuKZ1kZFhYfb1/eu
 1/eNXLA2sfevTXlX3qq/n5Mtbod9Ndn4+KbaYFQA=
X-Developer-Key: i=alchark@flipper.net; a=openpgp;
 fpr=9DF6A43D95320E9ABA4848F5B2A2D88F1059D4A5

HUSB311 is a pin-compatible and register-compatible drop-in replacement
for RT1711H, so add its compatible string to the existing binding.

Link: https://www.hynetek.com/uploadfiles/site/219/news/0863c0c7-f535-4f09-bacd-0440d2c21088.pdf
Link: https://dl.xkwy2018.com/downloads/RK3588S/03_Product%20Line%20Branch_Tablet/02_Key%20Device%20Specifications/HUSB311%20introduction%2020210526.pdf
Link: https://www.richtek.com/assets/product_file/RT1711H/DS1711H-04.pdf
Signed-off-by: Alexey Charkov <alchark@flipper.net>
Reviewed-by: Krzysztof Kozlowski <krzysztof.kozlowski@oss.qualcomm.com>
---
 Documentation/devicetree/bindings/usb/richtek,rt1711h.yaml | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/Documentation/devicetree/bindings/usb/richtek,rt1711h.yaml b/Documentation/devicetree/bindings/usb/richtek,rt1711h.yaml
index 210090308e7c..7ded36384518 100644
--- a/Documentation/devicetree/bindings/usb/richtek,rt1711h.yaml
+++ b/Documentation/devicetree/bindings/usb/richtek,rt1711h.yaml
@@ -22,12 +22,17 @@ properties:
       - enum:
           - richtek,rt1711h
           - richtek,rt1715
+      - items:
+          - enum:
+              - hynetek,husb311
+          - const: richtek,rt1711h
       - items:
           - enum:
               - etekmicro,et7304
           - const: richtek,rt1715
     description:
       RT1711H support PD20, ET7304 and RT1715 support PD30 except Fast Role Swap.
+      HUSB311 is a rebrand of RT1711H which is pin and register compatible.
 
   reg:
     maxItems: 1

From patchwork Wed Mar 18 14:32:56 2026
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Alexey Charkov <alchark@flipper.net>
X-Patchwork-Id: 14481762
Received: from mail-wm1-f42.google.com (mail-wm1-f42.google.com
 [209.85.128.42])
	(using TLSv1.2 with cipher ECDHE-RSA-AES128-GCM-SHA256 (128/128 bits))
	(No client certificate requested)
	by smtp.subspace.kernel.org (Postfix) with ESMTPS id 951B33DEACB
	for <linux-usb@vger.kernel.org>; Wed, 18 Mar 2026 14:33:24 +0000 (UTC)
Authentication-Results: smtp.subspace.kernel.org;
 arc=none smtp.client-ip=209.85.128.42
ARC-Seal: i=1; a=rsa-sha256; d=subspace.kernel.org; s=arc-20240116;
	t=1773844406; cv=none;
 b=gYXEV73BFzx6EMSFFhS4E/URhl/ADIESCJWdKwyxmY3Bd5YeruvC0dLjPwsZ9AtmP68kDwK1jGIg+8nyH2dm4z7FBd4lNcpTVjSDkflP3Zm5LzZehDZ/3lss3oo1QZI+ME0zGElwumBkWnTvv18GxyJuaCGDIqT2rYqKLX3GZ0A=
ARC-Message-Signature: i=1; a=rsa-sha256; d=subspace.kernel.org;
	s=arc-20240116; t=1773844406; c=relaxed/simple;
	bh=qSIoqe1rCM98pYxPjvfzxpcj4EO8iAA2yOd3l9ZnBNY=;
	h=From:Date:Subject:MIME-Version:Content-Type:Message-Id:References:
	 In-Reply-To:To:Cc;
 b=TdVrUeUMxZ7HEnbGqybNMSWHs+P4wdWGklHgKZjRxEBmLCe+MAFapm4LlYf7K+KIb1L9iN8R7sObfIMo7CB5xEx4uy8fc6Mthv8/j97TOg4qzfCln7PLOn0/fIwaUNyFUMmYedpUlE7kCrifAa6KUlUftEZf082+nxA0Te2hv/0=
ARC-Authentication-Results: i=1; smtp.subspace.kernel.org;
 dmarc=pass (p=quarantine dis=none) header.from=flipper.net;
 spf=pass smtp.mailfrom=flipper.net;
 dkim=pass (2048-bit key) header.d=flipper.net header.i=@flipper.net
 header.b=Pjpra7Of; arc=none smtp.client-ip=209.85.128.42
Authentication-Results: smtp.subspace.kernel.org;
 dmarc=pass (p=quarantine dis=none) header.from=flipper.net
Authentication-Results: smtp.subspace.kernel.org;
 spf=pass smtp.mailfrom=flipper.net
Authentication-Results: smtp.subspace.kernel.org;
	dkim=pass (2048-bit key) header.d=flipper.net header.i=@flipper.net
 header.b="Pjpra7Of"
Received: by mail-wm1-f42.google.com with SMTP id
 5b1f17b1804b1-48628ce9ab5so27373215e9.2
        for <linux-usb@vger.kernel.org>; Wed, 18 Mar 2026 07:33:24 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=flipper.net; s=google; t=1773844403; x=1774449203;
 darn=vger.kernel.org;
        h=cc:to:in-reply-to:references:message-id:content-transfer-encoding
         :mime-version:subject:date:from:from:to:cc:subject:date:message-id
         :reply-to;
        bh=D1QFYrzbzDmyd8ewHMMN3Hovj77cWoAajjF/dHg7uJs=;
        b=Pjpra7OfzTzWpcbxrCJaqgs07DRByztlLJFK6htNWU+bCyZWHot+TWc1NRPQbtPM1M
         rLKgSmHvwXFehTFMfbiKkrD+YOwtKmSSKuGBJVBjJUOZw/fIPcJALqvR7iG5oREHLwcq
         UD7Upi6CWmCPrHA7zx8KF05OVk0zzmfL0gyrX4vp4saFdyghcC1EQlaFXpu3U0g3X6qd
         jcCSvMSzTlPp0gufoUF9ma6hwFWgmZFX7QFPIkaE0qAR6qtjr4SimiVJ9MdTQsoluslr
         Z1GLqunH6Zo4WUC1y1OxmYKl2R+i8eV6YMEqiKgXhw3OcZeHgkrEKIrtJ2SjGwJrJwYs
         8AwQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20251104; t=1773844403; x=1774449203;
        h=cc:to:in-reply-to:references:message-id:content-transfer-encoding
         :mime-version:subject:date:from:x-gm-gg:x-gm-message-state:from:to
         :cc:subject:date:message-id:reply-to;
        bh=D1QFYrzbzDmyd8ewHMMN3Hovj77cWoAajjF/dHg7uJs=;
        b=rGieXwTbt57zUkul2LyvjsX/3O8jE0tSf+FOia/Mw8nyJ+D9AI8wRlOsoxam7BDJOK
         auRE9baF3lC1+nUSROcTmzGYgqcv4p11Zh656jPyNas4/p2NjnOcpHWiuTu4njo/XCcS
         mEPp0W3iJzO6vNE0oce42/LZZ9ldUyO6sak9Y7rqcB6rcfi9CBnWsjnksfm8AAa5hCs+
         sCs3ul+yqjcSYc9DPUgBgNkqdeCu4IM3wnfUNPQELCzPzxTPzHcJnEVbxG+NFyXRHtf0
         ysliUzaCnCbfQDFp5SKKX9tHkPAL6sniS7u4dk6JoMEw0Dp2gmtdd7wFy/0oufSktDrN
         +p2A==
X-Forwarded-Encrypted: i=1;
 AJvYcCXnGcD0EmnlQE8r/LlgDsEwjP7ryisuw7b4Rb+lslq77+dAlvV/H2fZbEwlCNI21Kwc6/0P7kRFXAc=@vger.kernel.org
X-Gm-Message-State: AOJu0Yy19DYbOBTL3hsZh2mmj6ZEGZop0occsM3fy+HpC9RXkBbgOp33
	0Rb+QgcDzJ1vfwiwCXzJMfAIxpXUbQMNWhWnv8pcBjNiWj0MgXqiXrYASFJdTHjMbNQ=
X-Gm-Gg: ATEYQzx40esjIcV4EUc5RyxWc8Ee+OIm/fmbMVO758HgWo/RU29q04UMONVI2cjBnyG
	s1z03KRVSeppBoaUaObZ3diTeEzN7BmSqyQ9C03d3GbVOqi0mTudClY+/IpTpex0j0ErghaAPAw
	DSqH31IdkHLe6jKghxUlepyNjeBR/KShYrYso0mUpUO1gtqmYQHeWmJTbPF9FLI9bAnI/hJ8Omn
	FZiJ2tZAlvRhxg1/YKR+r+xepyuTlL7LqdYLbudk1Ra4sPd8A5aSuFbJ468rBfapRLneVE+Ysa2
	7pETteM1IRwAWXr8e//Z+M5hK7AYqYO0CikqZN5Ka8RYbBCSXayYUPkX/6qjvUor2v+DteKrfhu
	VJX1Cm15FBU1ff9UY2lwTnv4uKEZ+b+AbBo28r6QmNa9xuxkXic1W1cFMgoAUgUqIxok/3RF2xr
	A9x7VQDteLACEco6ZfZUXEtDZOc3YfQnq1uS16qDElsHuVFB+Oq9ICgcJ/lq4uwxlHKLA3pVk4j
	CU/Sw==
X-Received: by 2002:a05:600c:8484:b0:485:35a4:939c with SMTP id
 5b1f17b1804b1-486f4476c41mr61407685e9.29.1773844402828;
        Wed, 18 Mar 2026 07:33:22 -0700 (PDT)
Received: from alchark-surface.localdomain (bba-86-98-192-109.alshamil.net.ae.
 [86.98.192.109])
        by smtp.gmail.com with ESMTPSA id
 5b1f17b1804b1-4856ea8fb0dsm270642085e9.3.2026.03.18.07.33.20
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Wed, 18 Mar 2026 07:33:22 -0700 (PDT)
From: Alexey Charkov <alchark@flipper.net>
Date: Wed, 18 Mar 2026 18:32:56 +0400
Subject: [PATCH v4 4/4] usb: typec: tcpci_rt1711h: Drop unnecessary
 VID/PID/DID checks
Precedence: bulk
X-Mailing-List: linux-usb@vger.kernel.org
List-Id: <linux-usb.vger.kernel.org>
List-Subscribe: <mailto:linux-usb+subscribe@vger.kernel.org>
List-Unsubscribe: <mailto:linux-usb+unsubscribe@vger.kernel.org>
MIME-Version: 1.0
Message-Id: <20260318-husb311-v4-4-69e029255430@flipper.net>
References: <20260318-husb311-v4-0-69e029255430@flipper.net>
In-Reply-To: <20260318-husb311-v4-0-69e029255430@flipper.net>
To: Rob Herring <robh@kernel.org>, Krzysztof Kozlowski <krzk+dt@kernel.org>,
 Conor Dooley <conor+dt@kernel.org>,
 Heikki Krogerus <heikki.krogerus@linux.intel.com>,
 Greg Kroah-Hartman <gregkh@linuxfoundation.org>,
 Gene Chen <gene_chen@richtek.com>, Yuanshen Cao <alex.caoys@gmail.com>
Cc: Sebastian Reichel <sebastian.reichel@collabora.com>,
 devicetree@vger.kernel.org, linux-kernel@vger.kernel.org,
 linux-usb@vger.kernel.org,
 Krzysztof Kozlowski <krzysztof.kozlowski@oss.qualcomm.com>,
 Alexey Charkov <alchark@flipper.net>
X-Mailer: b4 0.14.3
X-Developer-Signature: v=1; a=openpgp-sha256; l=3831; i=alchark@flipper.net;
 h=from:subject:message-id; bh=qSIoqe1rCM98pYxPjvfzxpcj4EO8iAA2yOd3l9ZnBNY=;
 b=owGbwMvMwCW2adGNfoHIK0sZT6slMWTu2r6EhS9H+IT/RWf5yla7/yWBwjP+XWvJzfjq9eLc1
 6SH07IudExkYRDjYrAUU2SZ+22J7VQjvlm7PDy+wsxhZQIZIi3SwAAELAx8uYl5pUY6Rnqm2oZ6
 hoY6xjpGDFycAjDVXtWMDKfaFCd8FpDq5AlsWHTjbmrM9maTVRGNL2VFuv0+rPp94hojw9Og62Z
 rXFvMkxXKxCfuPsVnt+BG+NHZP3OvPQvm2ODowgIA
X-Developer-Key: i=alchark@flipper.net; a=openpgp;
 fpr=9DF6A43D95320E9ABA4848F5B2A2D88F1059D4A5

Existing checks for VID/PID/DID in the driver are redundant since the
driver is already matched to the device via I2C device ID and OF
compatible strings, and they preclude the use of fallback compatibles.

Remove them to make the driver slimmer and adding new clones easier.

Reviewed-by: Heikki Krogerus <heikki.krogerus@linux.intel.com>
Signed-off-by: Alexey Charkov <alchark@flipper.net>
---
 drivers/usb/typec/tcpm/tcpci_rt1711h.c | 59 +-------------------------
 1 file changed, 2 insertions(+), 57 deletions(-)

diff --git a/drivers/usb/typec/tcpm/tcpci_rt1711h.c b/drivers/usb/typec/tcpm/tcpci_rt1711h.c
index 5f44accac9dd..0af95c2d33fc 100644
--- a/drivers/usb/typec/tcpm/tcpci_rt1711h.c
+++ b/drivers/usb/typec/tcpm/tcpci_rt1711h.c
@@ -18,13 +18,6 @@
 #include <linux/regmap.h>
 #include <linux/regulator/consumer.h>
 
-#define RT1711H_VID		0x29CF
-#define ET7304_VID		0x6DCF
-#define RT1711H_PID		0x1711
-#define RT1711H_DID		0x2171
-#define RT1715_DID		0x2173
-#define ET7304_DID		0x2173
-
 #define RT1711H_PHYCTRL1	0x80
 #define RT1711H_PHYCTRL2	0x81
 
@@ -57,8 +50,6 @@
 
 struct rt1711h_chip_info {
 	u32 rxdz_sel;
-	u16 vid;
-	u16 did;
 	bool enable_pd30_extended_message;
 };
 
@@ -304,35 +295,6 @@ static int rt1711h_sw_reset(struct rt1711h_chip *chip)
 	return 0;
 }
 
-static int rt1711h_check_revision(struct i2c_client *i2c, struct rt1711h_chip *chip)
-{
-	int ret;
-
-	ret = i2c_smbus_read_word_data(i2c, TCPC_VENDOR_ID);
-	if (ret < 0)
-		return ret;
-	if (ret != chip->info->vid) {
-		dev_err(&i2c->dev, "vid is not correct, 0x%04x\n", ret);
-		return -ENODEV;
-	}
-	ret = i2c_smbus_read_word_data(i2c, TCPC_PRODUCT_ID);
-	if (ret < 0)
-		return ret;
-	if (ret != RT1711H_PID) {
-		dev_err(&i2c->dev, "pid is not correct, 0x%04x\n", ret);
-		return -ENODEV;
-	}
-	ret = i2c_smbus_read_word_data(i2c, TCPC_BCD_DEV);
-	if (ret < 0)
-		return ret;
-	if (ret != chip->info->did) {
-		dev_err(&i2c->dev, "did is not correct, 0x%04x\n", ret);
-		return -ENODEV;
-	}
-	dev_dbg(&i2c->dev, "did is 0x%04x\n", ret);
-	return ret;
-}
-
 static void rt1711h_unregister_tcpci_port(void *tcpci);
 
 static int rt1711h_probe(struct i2c_client *client)
@@ -351,12 +313,6 @@ static int rt1711h_probe(struct i2c_client *client)
 
 	chip->info = i2c_get_match_data(client);
 
-	ret = rt1711h_check_revision(client, chip);
-	if (ret < 0) {
-		dev_err(&client->dev, "check vid/pid fail\n");
-		return ret;
-	}
-
 	chip->data.regmap = devm_regmap_init_i2c(client,
 						 &rt1711h_regmap_config);
 	if (IS_ERR(chip->data.regmap))
@@ -412,27 +368,16 @@ static void rt1711h_unregister_tcpci_port(void *tcpci)
 	tcpci_unregister_port(tcpci);
 }
 
-static const struct rt1711h_chip_info et7304 = {
-	.rxdz_sel = RT1711H_BMCIO_RXDZSEL,
-	.vid = ET7304_VID,
-	.did = ET7304_DID,
-	.enable_pd30_extended_message = true,
-};
-
 static const struct rt1711h_chip_info rt1711h = {
-	.vid = RT1711H_VID,
-	.did = RT1711H_DID,
 };
 
 static const struct rt1711h_chip_info rt1715 = {
 	.rxdz_sel = RT1711H_BMCIO_RXDZSEL,
-	.vid = RT1711H_VID,
-	.did = RT1715_DID,
 	.enable_pd30_extended_message = true,
 };
 
 static const struct i2c_device_id rt1711h_id[] = {
-	{ "et7304", (kernel_ulong_t)&et7304 },
+	{ "et7304", (kernel_ulong_t)&rt1715 },
 	{ "rt1711h", (kernel_ulong_t)&rt1711h },
 	{ "rt1715", (kernel_ulong_t)&rt1715 },
 	{}
@@ -440,7 +385,7 @@ static const struct i2c_device_id rt1711h_id[] = {
 MODULE_DEVICE_TABLE(i2c, rt1711h_id);
 
 static const struct of_device_id rt1711h_of_match[] = {
-	{ .compatible = "etekmicro,et7304", .data = &et7304 },
+	{ .compatible = "etekmicro,et7304", .data = &rt1715 },
 	{ .compatible = "richtek,rt1711h", .data = &rt1711h },
 	{ .compatible = "richtek,rt1715", .data = &rt1715 },
 	{}
